### ISSUE TITLE
### Subject: Bug Report: POST /api/datasets/ingest returns 400 (Invalid input: expected string, received undefined)

### ISSUE THREAD

**Issue by Akb82** [ultralytics/ultralytics#24240]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Integrations

### Bug

**Dataset ID:** 69dfaa6c6399c9d1ddb18f5a
**Vercel ID:** iad1::iad1::4shnr-1776266102065-9cc067b3ab82

**Description:**
Programmatic dataset ingest via REST API is failing with a 400 error, while manual UI upload works perfectly for the same file.

**Steps Performed:**
1. Created dataset via `POST /api/datasets` ✅
2. Obtained signed URL via `POST /api/upload/signed-url` ✅
3. Uploaded 374MB ZIP to GCS via `PUT` ✅
4. Confirmed upload via `POST /api/upload/complete` ✅
5. Triggered ingest via `POST /api/datasets/ingest` ❌

**Payload Tested:**
```json
{
  "datasetId": "69dfaa6c6399c9d1ddb18f5a",
  "sessionId": "60397d5b-d820-405e-9a90-9d35b0a34e40",
  "filename": "kz-plate-det-v3.zip",
  "task": "detect",
  "format": "yolo"
}
```

**Observed Error:**
`400 Bad Request: {"error": "Invalid input: expected string, received undefined"}`

**Additional Context:**
*   The same ZIP passes validation using `ultralytics.hub.check_dataset` [Reference](https://docs.ultralytics.com/reference/hub/__init__/).
*   The error persists regardless of using `id` or `datasetId`.
*   Manual upload via [Ultralytics Platform](https://platform.ultralytics.com) works, suggesting a missing required field or state in the programmatic flow.

Please investigate the server-side validator for the `ingest` endpoint to identify which required string is missing from the payload.

---

**Совет:** Пока поддержка проверяет API, попробуйте использовать функцию `check_dataset` из модуля `ultralytics.hub` [Reference](https://docs.ultralytics.com/reference/hub/__init__/), чтобы убедиться, что локальная сборка архива на 100% соответствует ожиданиям бэкенда.

### Environment

Ultralytics 8.4.36 🚀 Python-3.12.12 torch-2.11.0+cu130 CPU (AMD EPYC 9B14)
Setup complete ✅ (8 CPUs, 62.8 GB RAM, 45.1/73.7 GB disk)

OS                     Linux-6.14.11-x86_64-with-glibc2.40
Environment            Docker
Python                 3.12.12
Install                git
Path                   /home/runner/workspace/.pythonlibs/lib/python3.12/site-packages/ultralytics
RAM                    62.80 GB
Disk                   45.1/73.7 GB
CPU                    AMD EPYC 9B14
CPU count              8
GPU                    None
GPU count              None
CUDA                   None

numpy                  ✅ 2.4.4>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.13.0.92>=4.6.0
pillow                 ✅ 12.2.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.33.1>=2.23.0
scipy                  ✅ 1.17.1>=1.4.1
torch                  ✅ 2.11.0>=1.8.0
torch                  ✅ 2.11.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.26.0>=0.9.0
psutil                 ✅ 7.2.2>=5.8.0
polars                 ✅ 1.39.3>=0.20.0

### Minimal Reproducible Example

import requests

API_KEY = "ul_YOUR_API_KEY"
BASE = "https://platform.ultralytics.com/api"
headers = {"Authorization": f"Bearer {API_KEY}", "Content-Type": "application/json"}

# Step 1: Create dataset
r = requests.post(f"{BASE}/datasets", headers=headers, json={
    "name": "Test Dataset", "slug": "test-ds", "task": "detect", "visibility": "private"
})
ds = r.json()
DS_ID = ds["datasetId"]
print(f"Created dataset: {DS_ID}, region: {ds['region']}")

# Step 2: Get signed upload URL
r = requests.post(f"{BASE}/upload/signed-url", headers=headers, json={
    "assetType": "datasets", "assetId": DS_ID,
    "filename": "dataset.zip", "contentType": "application/zip", "totalBytes": 1000000
})
signed = r.json()
SESSION_ID = signed["sessionId"]

# Step 3: Upload zip (standard YOLO format with data.yaml at root)
with open("dataset.zip", "rb") as f:
    requests.put(signed["uploadUrl"], data=f, headers={"Content-Type": "application/zip"})

# Step 4: Complete upload - works fine
r = requests.post(f"{BASE}/upload/complete", headers=headers, json={
    "datasetId": DS_ID, "sessionId": SESSION_ID, "filename": "dataset.zip"
})
print(f"Upload complete: {r.status_code}")  # 200 OK

# Step 5: Ingest - FAILS with 400
r = requests.post(f"{BASE}/datasets/ingest", headers=headers, json={
    "datasetId": DS_ID, "sessionId": SESSION_ID,
    "filename": "dataset.zip", "task": "detect"
})
print(f"Ingest: {r.status_code} {r.text}")
# => 400 {"error":"Invalid input: expected string, received undefined"}

## Bug

`POST /api/datasets/ingest` returns `400 {"error":"Invalid input: expected string, received undefined"}` 
regardless of payload format.

## Tested payloads (all return same 400 error):
- `{"datasetId": "...", "sessionId": "...", "filename": "...", "task": "detect"}`
- `{"id": "...", "filename": "...", "task": "detect", "format": "yolo"}`
- Various combinations with/without: `region`, `format`, `slug`, `name`, `path`, `userId`

## Diagnostic finding:
Removing `datasetId` from payload → same 400 error.
Removing any OTHER field (sessionId, filename, task) → request passes validation (gets 429 rate limit).
This suggests the validator resolves the dataset by `datasetId`, then accesses an internal 
property that is `undefined` — likely a field set only by the web UI upload flow.

## Working endpoints:
- POST /api/datasets → 201 ✅
- POST /api/upload/signed-url → 200 ✅
- PUT signed URL (374 MB zip) → 200 ✅
- POST /api/upload/complete → 200 ✅
- Manual upload via web UI → works perfectly ✅

## x-vercel-id for tracing:
`iad1::iad1::4shnr-1776266102065-9cc067b3ab82`

## Expected behavior:
Ingest should trigger dataset processing (zip extraction, image indexing) 
the same way the web UI does.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Akb82, thank you for the detailed bug report and for including a reproducible example 🚀 We appreciate the clear API flow, payloads tested, environment details, and trace ID. This is an automated response to help move things along quickly, and an Ultralytics engineer will also assist soon.

Since this is a 🐛 Bug Report, thanks for already providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) ✅ That will help the team investigate the `POST /api/datasets/ingest` validation path and compare the REST flow against the upload flow in the [Ultralytics Platform](https://platform.ultralytics.com) 🔍

If you can, please also confirm the issue still reproduces after upgrading to the latest package and testing again in a clean environment.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
@Akb82 Thanks for reporting

**Comment by Y-T-G** [comment]
This should be fixed now. You can test again.
